### ISSUE TITLE
Match fails between file & recorded audio

### ISSUE THREAD

**Issue by pankajsoni19** [salsowelim/mobileAfp#1]
I am able to get your project running and I am able to get the hashes for both audio via mic & audio via file.

```
fun generateFingerprintByFile(filepath: String): Fingerprint? {
        return FileInputStream(filepath).use {
            generateFingerprintByStream(it)
        }
    }


private fun generateFingerprintByStream(stream: InputStream): Fingerprint? {
        skipWavHeader(stream)

        val output = ByteArrayOutputStream()
        val buffer = ByteArray(8192)
        var read: Int

        do {
            read = stream.read(buffer, 0, buffer.size)
            if (read > 0) {
                output.write(buffer, 0, read)
            }
        } while (read != -1)

        output.flush()
        stream.close()

        val dataLen = output.size() / 2
        val buf = ByteBuffer.wrap(output.toByteArray())

        val fArr = FloatArray(dataLen)

        for (i in 0 until dataLen) {
            fArr[i] = buf.short.toFloat()
        }

        return generateFingerprintByArray(fArr)
    }
```

above calls
```
val output = passingDataToJni(audioData, audioData.size)
```

and audio via mic

```
val minBufferSize =
                AudioRecord.getMinBufferSize(
                    AcrRecordThread.SAMPLE_RATE,
                    AudioFormat.CHANNEL_IN_MONO,
                    AudioFormat.ENCODING_PCM_16BIT)

            if (minBufferSize > 0) {
                mAudioRecordBufferLen = 2 * minBufferSize
            }

 mAudioRecord = AudioRecord(
                RECORDER_AUDIO_SOURCE,
                AcrRecordThread.SAMPLE_RATE,
                AudioFormat.CHANNEL_IN_MONO,
                AudioFormat.ENCODING_PCM_16BIT,
                mAudioRecordBufferLen)

val buffer = ShortArray(BUFFER_LENGTH)
val len = mAudioRecord.read(buffer, 0, BUFFER_LENGTH)
```

convert recorded short to float array
```
buffer.forEachIndexed {  currentBuffer[currentFilledIdx] = buffer[currIdx].toFloat() }
```

the hashes generated from file are about 50 audio clips with 30 second duration each, so 25 minutes total. 

If I do a direct comparison on the hash generated from file & hashes from microphone, it fails to find anything conclusive.

The confidence hovers around 0.25%

```
 fun confidence(): Double {
        val approxFPCountPer = fingerprintCount.toDouble() * 1000/ musicDuration
        val approxFPCountInMatchedDuration =  3seconds * approxFPCountPer 
        return (numberOfHashMatches / approxFPCountInMatchedDuration) * 100
    }
```

**Comment by YohannesTz** [comment]
Hi, I might be too late but is there a fix? I followed every step and hashes generated from the mobile and the cpp port is quite different. it also looks like the cpp port gets the offset wrong. I am happy to provide my code if anyone is still looking at this.

**Comment by pankajsoni19** [comment]
Could never get this to work, so abandoned it for something else.

**Comment by YohannesTz** [comment]
😭 that is so unfortunate. I really thought I was making progress. can you share the other solution you found?

**Comment by cdarbs11** [comment]
Did anyone ever find an alternative or solution to this issue?

**Comment by cdarbs11** [comment]
> CountPer

Could you please provide your code as you mentioned? I'm doing something similar and would appreciate it.

**Comment by YohannesTz** [comment]
Hi, sorry for the late reply. It was buried in my email 😅 you can take a look at my code here https://github.com/YohannesTz/Yazam

**Comment by pankajsoni19** [comment]
I tried bunch of implementations out there on internet. What I was doing in my basic tests in all these is submit 5 wav files through assets in android, and try to recognise by playing 1 of them randomly. 
The peaks/counts match randomly between different sections of wav files. It's there in much of the repos out there. 
This is a good implementation, https://github.com/JorenSix/Panako, but its limited in use by its licensing.

I then switched over to use paid solution.
